src/frontend/stores/component-data/component-data-store.ts

```diff
@@ -97,6 +97,45 @@ export class ComponentDataStore extends AbstractStore {
   }
 
   /**
+   * Get the nodes from root of the tree to given node.
+   * @param  {Object} node
+   */
+  private getPathToNode(node: any): any[] {
+    // The ID shows the path from root to given node.
+    let splitId: number[] = node.id.split('.');
+    let nodes: any[] = [];
+
+    // First index is for the component root:
+    let rootIndex: number = splitId.shift();
+    let root = this._componentData[rootIndex];
+    nodes.push(root);
+
+    // Next traverse the tree by index from ID.
+    splitId.reduce(function (last: any, subId: number) {
+      let nextChild = last.children[subId];
+      nodes.push(nextChild);
+      return nextChild;
+    }, root);
+    return nodes;
+  }
+
+  /**
+   * Opens the nodes from root of the tree to given node
+   * to reveal the given node.
+   * @param  {Object} node
+   */
+  private openPathToNode(node: any) {
+    let path = this.getPathToNode(node);
+    path.pop(); // Only open until the parent of selected.
+
+    path.forEach((n) => {
```

Comment:
There's a lint warning for shadowed variables since `node` is a param in `openPathToNode`. I can rename this from `n` to  `pathNode`